register-event.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh -l
+
+export PATH=$PATH:/app/cmd/pipectl
+
+if [ -z "$5" ]
+then
+    pipectl event register \
+        --address=$1 \
+        --api-key=$2 \
+        --name=$3 \
+        --data=$4
+else
+    pipectl event register \
+        --address=$1 \
+        --api-key=$2 \
+        --name=$3 \
+        --data=$4 \
+        --labels=$5
+fi
```

Comment:
I know it isn't classy but I thought that is much clearer than doing something to make it shorter.

Reply:
If someone is aware, let me know.

Reply:
I think using `$#` to check the number of arguments is better to use in L5 - if condition 🤔

Reply:
In that case, it's likely to catch the case of missing other args (e.g. missing `$4` case goes on the `then` statement), right?

Reply:
Yeh, or using `case` statement 🤔 In that case, this `pipectl` command only runs if required arguments are all passed.

Reply:
Because I can't say i'm enough familiar with shell script, I can't even imagine how it can make it elegant. Could you show me a short example?

Reply:
sorry for the late 🙏 It would look like this
```bash
case "$#" in
	4 )
		echo "4 arguments were passed"
		;;
	5 )
		echo "5 arguments were passed"
		;;
	* )
		echo "out of valid"
		;;
esac
```
check
```
╭─ ~/workspace/works/test 
╰─ ./sample.sh 1 2 3
out of valid

╭─ ~/workspace/works/test 
╰─ ./sample.sh 1 2 3 4
4 arguments were passed

╭─ ~/workspace/works/test 
╰─ ./sample.sh 1 2 3 4 5
5 arguments were passed
```

Reply:
Thank you for sharing the simplest examle! But you know what? I have to tell you something.
I don't know exactly because I haven't tested yet, but from skimming [the documentation](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action#creating-an-action-metadata-file), seems like all of inputs will get given to the docker container. If $5 wan't given, then maybe just "" will get populated. The point is, "$#" is still 5.

```yaml
runs:
  args:
    - ${{ inputs.api-address }} => $1
    - ${{ inputs.api-key }} => $2
    - ${{ inputs.event-name }} => $3
    - ${{ inputs.data }} => $4
    - ${{ inputs.labels }} => $5
```

Reply:
> seems like all of inputs will get given to the docker container

I see, thanks for sharing 🙏 Then I get we still have this `-z` check here as is 👍